### PR TITLE
updating group type annotation for netcdf, hdf5, and zarr open_datatree function

### LINKED ISSUE #9597: open_datatree `group` parameter does not support Iterable[str]
### What is your issue?

The annotation typing for the  `group` parameter in the `open_datatree` function receives `str | Iterable[str] | Callable | None = None` for all backends. for example

https://github.com/pydata/xarray/blob/f01096fef402485092c7132dfd042cc8f467ed09/xarray/backends/zarr.py#L1285

However, the current implementation does not support `Iterable[str]`; Therefore, it should be removed in all backends

### PR DESCRIPTION
**Pull request by aladinor**
- [X] Closes #9597
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/backends/h5netcdf_.py [hunk 1]
```diff
@@ -465,7 +465,7 @@ def open_datatree(
         use_cftime=None,
         decode_timedelta=None,
         format=None,
-        group: str | Iterable[str] | Callable | None = None,
+        group: str | Callable[[], str] | None = None,
```
Comment (headtr1ck) [inline xarray/backends/h5netcdf_.py:468]: Is that really the correct Callable signature? Does Callable actually work at all?

**Review by headtr1ck — APPROVED** [review]
Thanks for the quick PR!

**Comment by headtr1ck** [comment]
Feel free to add an entry to whats-new!

File: xarray/backends/h5netcdf_.py (hunk 1 above)
Comment (aladinor) [reply xarray/backends/h5netcdf_.py:468]: I don't think so. However, this is something I will discuss in the next datatree meeting.

Comment (keewis) [reply xarray/backends/h5netcdf_.py:468]: `Callable` was a suggestion I had some time ago, but it also is unsupported right now. So I'd also remove that, and possibly add it back when/if we add support for `Iterable[str]`.

**Review by TomNicholas — APPROVED** [review]

**Pull request merged**